### PR TITLE
Adding support for JSON string values in Cosmos DB Trigger binding

### PR DESCRIPTION
**Pull request by ealsur**
This change enables executing the Trigger with a test payload from the Functions UI.

### CODE REVIEW

File: src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs [hunk 1]
```diff
@@ -84,5 +81,28 @@ public ParameterDescriptor ToParameterDescriptor()
                 CollectionName = this._documentCollectionLocation.CollectionName
             };
         }
+
+        internal static bool TryAndConvertToDocumentList(object value, out IReadOnlyList<Document> documents)
+        {
+            try
+            {
+                string valueAsString = value as string;
```
Comment (mamaso) [inline src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs]: small nit with C# 7 syntax:

```
if (value is IReadOnlyList<Document> docs) 
{
  documents = docs;
}
else if (value is string stringVal)
{
...
}
```

Comment (ealsur) [reply src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs]: @mamaso Thanks! I sometimes forget I can use C#7 in these repos 😄

**Comment by ealsur** [comment]
@mamaso @MikeStall can this PR be reviewed? Are any changes required for it to be merged?

File: src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs [hunk 2]
```diff
@@ -84,5 +81,31 @@ public ParameterDescriptor ToParameterDescriptor()
                 CollectionName = this._documentCollectionLocation.CollectionName
             };
         }
+
+        internal static bool TryAndConvertToDocumentList(object value, out IReadOnlyList<Document> documents)
+        {
+            try
```
Comment (mamaso) [inline src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs]: nit: 

```c#
documents = null;
try
{
...
// remove else block as documents defaults to null
}
finally
{
  return documents != null;
}
```

Comment (ealsur) [reply src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs]: @mamaso Tried implementing it but I get a "Control cannot leave the body of a finally clause"

Comment (ealsur) [reply src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerBinding.cs]: I removed the else though

**Comment by ealsur** [comment]
@mamaso @MikeStall Think this can be merged?

**Comment by fabiocav** [comment]
@ealsur kicked off a full CI build. Will merge when that is done.

**Pull request merged**